modules/material-management/test/algorithm_suites.test.ts

```diff
@@ -28,10 +28,22 @@ describe('AlgorithmSuiteIdentifier', () => {
 describe('AlgorithmSuite', () => {
   it('should not allow an instance', () => {
     // @ts-ignore Trying to test something that Typescript should deny...
-    expect(() => new AlgorithmSuite()).to.throw()
+    expect(() => new AlgorithmSuite()).to.throw('new AlgorithmSuite is not allowed')
```

Comment:
Another niave JS question, what is going on here: https://github.com/awslabs/aws-encryption-sdk-javascript/blob/master/modules/material-management/src/algorithm_suites.ts#L105

Is this preventing the AlgorithmSuite constructor from being called? Is this because you're trying to ensure that only extensions of AlgorithmSuite are actually constructed?

Reply:
> Is this preventing the AlgorithmSuite constructor from being called? 

yes

> Is this because you're trying to ensure that only extensions of AlgorithmSuite are actually constructed?

yes

when you create an object with `new` there is a special property `constructor` that is just what it says.  :)